### DISCUSSION TITLE
Getting external submodules fails with cee-gitlab.sandia.gov port 22: Connection timed out

### DISCUSSION THREAD

**Discussion by loopless** [snl-dakota/dakota#93] [Q&A] [answered]
We have a build system based around Conan that used to download the tarball from "sandia.gov".  Then built from the extracted tar to create a Conan package.

Of course that is all gone now and we have to get source from GitHub. 

The git clone command

git clone https://github.com/snl-dakota/dakota.git --branch 6.18 --recurse-submodules

Fails on many external submodules with....

ssh: connect to host cee-gitlab.sandia.gov port 22: Connection timed out
fatal: Could not read from remote repository.

Are there are COMPLETE tarballs (including external packages like JEGA) out there for specific versions?

**Comment by briadam** [comment, marked as answer]
Direct answer: Yes there are complete source packages. See, e.g., https://github.com/snl-dakota/dakota/releases/tag/v6.19.0 linked from https://dakota.sandia.gov/downloads/.

If you want to git clone, you'll have to be selective about which submodules are initialized. Info here if helpful: https://snl-dakota.github.io/docs/6.19.0/users/compiling/getsource.html#getsource

**Reply by loopless** [reply]
The only issue is that the .tar.gz from that location does not include any of the submodules at all. I would not call that a "complete source package". But as you point out, picking the submodules to avoid the non-public ones should allow me to clone what I need using git commands.

**Comment by briadam** [comment]
I apologize, this confusion has come up before. The packages labeled "Source code" are auto-generated by a Github process and are indeed deficient in that way. You'll likely want 
[dakota-6.19.0-public-src-cli.tar.gz](https://github.com/snl-dakota/dakota/releases/download/v6.19.0/dakota-6.19.0-public-src-cli.tar.gz) where the CLI is for command-line interface. I just verified that's complete as expected.

**Reply by loopless** [reply]
Ah, yes, I see that works. Thanks.
I don't suppose there are older versions ( 6.13) available? 
Although I would guess 6.19 won't give me too many problems with changes since 6.13.
